### PR TITLE
Add macro `govukAttributes()` with `params.attributes` string support

### PR DESCRIPTION
**Pull request by colinrotherham**
This PR moves all attribute loops into a new `govukAttributes()` macro to close https://github.com/alphagov/govuk-frontend/issues/4666

It also supports `params.attributes` as strings to resolve [#4566 (comment)](https://github.com/alphagov/govuk-frontend/pull/4566#discussion_r1426926800)

## Before

Components use object-only attribute loops with no support for [boolean attributes](https://developer.mozilla.org/en-US/docs/Glossary/Boolean/HTML), e.g. `hidden` and `checked`

```njk
<strong class="govuk-tag {%- if params.classes %} {{ params.classes }}{% endif %}"
  {%- for attribute, value in params.attributes %} {{ attribute }}="{{ value }}"{% endfor %}>
```

## After

Components use new `govukAttributes()` macro supporting already-stringified and [boolean attributes](https://developer.mozilla.org/en-US/docs/Glossary/Boolean/HTML)

```njk
<strong class="govuk-tag {%- if params.classes %} {{ params.classes }}{% endif %}"
  {{- govukAttributes(params.attributes) }}>
```

### CODE REVIEW

**Review by romaricpascal — COMMENTED** [review]
Helper looks neat 🙌🏻 Regarding the handling of boolean values, is it a pressing issue to solve? If not, I'd be keen to not have it in this PR and plan a breaking change in the next major version. That could open the door to handling these boolean attributes as proper booleans as well (`true` they show, `false` they don't) 😊

**Comment by colinrotherham** [comment]
No not a pressing issue at all, wasn't sure if it was something your attribute config in https://github.com/alphagov/govuk-frontend/issues/2917 was aiming to solve?

Could quite easily wrap up two issues in one go

For example, use a config to flag exactly which attributes are optional—avoiding breaking changes

Otherwise you can see in https://github.com/alphagov/govuk-frontend/commit/b1bb9e81135cc5ff0d35c118d03d7d180c766e8a we can just turn off the HTML validate rule instead

It was only the Password input spike (button with `hidden` by default) that revealed the need for it

File: packages/govuk-frontend/src/govuk/macros/attributes.njk [hunk 1]
```diff
@@ -11,7 +11,12 @@
   {#- Append attribute name/value pairs -#}
   {%- if attributes is mapping %}
     {% for name, value in attributes %}
-      {% set attributesHtml = attributesHtml + " " + name | escape + '="' + value | escape + '"' %}
+      {#- Skip optional attribute value, e.g. `<input type="radio" checked>` -#}
+      {% if value === null or value === undefined %}
+        {% set attributesHtml = attributesHtml + " " + name | escape %}
```
[the remark this answers is not in the document]
Comment (colinrotherham) [reply packages/govuk-frontend/src/govuk/macros/attributes.njk]: Split out into https://github.com/alphagov/govuk-frontend/pull/4743

**Comment by colinrotherham** [comment]
@romaricpascal Split out boolean attributes into https://github.com/alphagov/govuk-frontend/pull/4743

So this one's ready to review again

**Pull request merged**